check/src/unify_type.rs

```diff
@@ -180,16 +182,22 @@ fn do_zip_match<'a, U>(self_: &ArcType,
                 }
                 Less => {
                     let offset = r_args.len() - l_args.len();
-                    let new_type =
-                        unifier.try_match(l, &Type::app(r.clone(), r_args[..offset].into()));
+
+                    let reduced_r = Type::app(r.clone(),
+                                              r_args[..offset].iter().cloned().collect());
```

Comment:
Could you make this an intermediate? like:

```rust
let offset = r_args.len() - l_args.len();

let reduced_r_args = r_args[..offset].iter().cloned();
let reduced_r = Type::app(r.clone(), reduced_r_args.collect());
```

Just find the weird indentation kinda ugly.